src/persistence.js

```diff
@@ -69,10 +70,14 @@ function createStorageWrapper(storage, transformers = []) {
   };
 
   const deserialize = (data) => {
-    const result =
+    const storageData =
       storage === localStorage() || storage === sessionStorage()
         ? JSON.parse(data).data
         : data;
+
+    const hasMigrations = Object.keys(migrations).length > 0;
+    const result = hasMigrations ? migrate(storageData, migrations) : storageData;
```

Comment:
It makes sense to run migrations before additional transformations occur since migrations are structural, where transformations operate on the leafs